packages/remix/src/utils/routerLoader.ts

```diff
@@ -0,0 +1,51 @@
+import { loadModule, logger } from '@sentry/utils';
+import { cwd } from 'process';
+import { DEBUG_BUILD } from './debug-build';
+import type { ReactRouterDomPkg } from './vendor/types';
+
+function hasMatchRoutes(pkg: ReactRouterDomPkg | undefined): boolean {
+  return !!pkg && typeof pkg.matchRoutes === 'function';
+}
+
+/**
+ * Loads the router package that provides matchRoutes.
+ * Tries loading @remix-run/router first, then falls back to react-router-dom.
+ */
+export async function loadRemixRouterModule(): Promise<ReactRouterDomPkg | undefined> {
+  // Try loading @remix-run/router first, then fall back to react-router-dom
+  for (const moduleName of ['@remix-run/router', 'react-router-dom']) {
+    const pkg = await tryLoadRouterModule(moduleName);
+
+    if (pkg) {
+      return pkg;
+    }
+  }
+
+  DEBUG_BUILD && logger.warn('Could not find a router package that provides `matchRoutes`.');
+
+  return;
+}
+
+async function tryLoadRouterModule(moduleName: string): Promise<ReactRouterDomPkg | undefined> {
+  let pkg: ReactRouterDomPkg | undefined;
+
+  pkg = loadModule<ReactRouterDomPkg>(moduleName);
+
+  if (hasMatchRoutes(pkg)) {
+    return pkg;
+  }
+
+  try {
+    pkg = await import(moduleName);
+  } catch (e) {
+    pkg = await import(`${cwd()}/node_modules/${moduleName}`);
```

Comment:
this doesn't work in CJS right? Can you confirm that?

Reply:
I have done a little research, but could not produce a case where this breaks. It looks like it works fine on both CJS and ESM. Not in worker environments though. 

But, while checking the previous discussions (https://github.com/getsentry/sentry-javascript/pull/5897), the main reason we need this was our TS version and it's up to date now, so we can just try making `@remix-run/router` as a dependency and stop doing this anyways? WDYT?

Reply:
> so we can just try making @remix-run/router a dependency and stop doing this anyways? WDYT

Let's do that instead!! Less dynamic imports the better.